internal/cluster/validator.go

```diff
@@ -224,6 +224,36 @@ func (v *clusterValidator) isApiVipValid(c *clusterPreprocessContext) Validation
 	return boolValue(err == nil)
 }
 
+func (v *clusterValidator) isNetworkTypeValid(c *clusterPreprocessContext) ValidationStatus {
+	validNetworkTypes := []string{models.ClusterNetworkTypeOVNKubernetes, models.ClusterNetworkTypeOpenShiftSDN}
+	if !funk.ContainsString(validNetworkTypes, swag.StringValue(c.cluster.NetworkType)) {
+		return ValidationFailure
+	}
+	if network.IsIPv6CIDR(c.cluster.ClusterNetworkCidr) && swag.StringValue(c.cluster.NetworkType) != models.ClusterNetworkTypeOVNKubernetes {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func (v *clusterValidator) printIsNetworkTypeValid(context *clusterPreprocessContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return "The cluster has a sufficient network type."
+	case ValidationFailure:
+		validNetworkTypes := []string{models.ClusterNetworkTypeOVNKubernetes, models.ClusterNetworkTypeOpenShiftSDN}
+		if !funk.ContainsString(validNetworkTypes, swag.StringValue(context.cluster.NetworkType)) {
+			return "NetworkType is not valid, valid network types can be OpenShiftSDN or OVNKubernetes"
+		}
+		if network.IsIPv6CIDR(context.cluster.ClusterNetworkCidr) && swag.StringValue(context.cluster.NetworkType) != models.ClusterNetworkTypeOVNKubernetes {
+			return "ipv6 can only use OVNKubernetes network type."
+		} else {
+			return "Network type is undefined"
```

Comment:
What does that mean?

Reply:
the validation failed for an unclear reason
changed "Network type is invalid", will that work?